operational_analysis/methods/electrical_losses_uq.py

```diff
@@ -0,0 +1,252 @@
+# This class defines key analytical routines for calculating electrical losses for 
+# a wind plant using operational data. Electrical loss is calculated per month and on 
+# an average annual basis by comparing monthly energy production from the turbines
+# and the revenue meter
+
+from operational_analysis import logged_method_call
+from operational_analysis import logging
+import numpy as np
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+
+class ElectricalLosses(object):
+    """
+    A serial (Pandas-driven) implementation of calculating the average monthly and annual
+    electrical losses at a wind plant, and their uncertainty. 
+    Energy output from the turbine SCADA meter and the wind plant revenue 
+    meter are used to estimate electrical losses.
+    
+    The approach is to first calculate daily sums of turbine and revenue meter energy over the 
+    plant period of record. Only those days where all turbines and the revenue meter were 
+    reporting for all timesteps are considered. Electrical loss is then the difference in 
+    total turbine energy production and meter production over those concurrent days. 
+    
+    A Monte Carlo approach is applied to sample revenue meter data and SCADA data
+    with a 0.5% imposed uncertainty, and one filtering parameter is sampled too.
+    The uncertainty in estimated electrical losses is quantified as standard deviation
+    of the distribution of losses obtained from the MC sampling.
+    
+    In the case that meter data is not provided on a daily or sub-daily basis (e.g. monthly), a
+    different approach is implemented. The sum of daily turbine energy is corrected for any missing 
+    reported energy data from the turbines based on the ratio of expected number of data counts per day 
+    to the actual. Daily corrected sum of turbine energy is then summed on a monthly basis. Electrical 
+    loss is then the difference between total corrected turbine energy production and meter production 
+    over those concurrent months.   
+    """ 
+
+    @logged_method_call
+    def __init__(self, plant, num_sim, uncertainty_meter=0.005, uncertainty_scada=0.005,
+                 uncertainty_correction_thresh=(0.9,0.995)):
+        """
+        Initialize electrical losses class with input parameters
+
+        Args:
+         plant(:obj:`PlantData object`): PlantData object from which EYAGapAnalysis should draw data.
+
```

Comment:
the new UQ-related args should be described here in the docstring

Reply:
I will add the following description:

        """
        Initialize electrical losses class with input parameters

        Args:
         plant(:obj:`PlantData object`): PlantData object from which EYAGapAnalysis should draw data.
         num_sim:(:obj:`int`): number of Monte Carlo simulations
         uncertainty_meter(:obj:`float`): uncertainty imposed to revenue meter data
         uncertainty_scada(:obj:`float`): uncertainty imposed to scada data
         uncertainty_correction_threshold(:obj:`float`): The data availability threshold (fraction) 
                                                         under which months should be eliminated (default is 0.95)
        """